web/libs/editor/src/components/Comments/Comment/CommentItem.tsx

```diff
@@ -42,198 +45,227 @@ interface CommentItemProps {
-              value={text}
-              onSubmit={async (value) => {
-                await updateComment(value);
-                setText(value);
-                await listComments({ suppressClearComments: true });
-              }}
-            />
-          ) : isConfirmDelete ? (
-            <Elem name="confirmForm">
-              <Elem name="question">Are you sure?</Elem>
-              <Elem name="controls">
-                <Button onClick={() => deleteComment()} size="compact" look="danger" autoFocus>
-                  Yes
-                </Button>
-                <Button onClick={() => setConfirmMode(false)} size="compact">
-                  No
-                </Button>
-              </Elem>
+        );
+      return null;
+    };
+
+    return (
+      <Block
+        name="comment-item"
+        mod={{ resolved, highlighted: isHighlighted }}
+        onMouseEnter={() => {
+          setHighlighted(true);
+        }}
+        onMouseLeave={() => {
+          setHighlighted(false);
+        }}
+        ref={_commentRef}
+      >
+        <Space spread size="medium" truncated>
+          <Space size="small" truncated>
+            <Elem tag={Userpic} user={createdBy} name="userpic" showUsername username={createdBy} />
+            <Elem name="name" tag="span">
+              {userDisplayName(createdBy)}
             </Elem>
-          ) : (
-            <>
-              {classifications?.default?.values?.length > 0 && (
-                <Elem name="classifications" tag="ul">
-                  {classifications?.default?.values?.map((valueArray: string[], index: number) => (
-                    <li key={index}>{valueArray.join("/")}</li>
-                  ))}
-                </Elem>
-              )}
-              {text}
-              {hasLinkState && (
-                <Elem name="linkState">
-                  <LinkState linking={linking} region={region} result={result} interactive />
+          </Space>
+
+          <Space size="small">
+            <Elem name="resolved" component={IconCheck} />
+            <Elem name="saving" mod={{ hide: isPersisted }}>
+              <Elem name="dot" />
+            </Elem>
+            <TimeTracker />
+          </Space>
+        </Space>
+
+        <Elem name="content">
+          <Elem name="text">
+            {isEditMode ? (
+              <>
+                <CommentFormBase
+                  value={text}
+                  onSubmit={async (value) => {
+                    await updateComment(value, classifications);
+                    setText(value);
+                    await listComments({ suppressClearComments: true });
+                  }}
+                  classifications={classifications}
+                />
+                {classificationsItems.length > 0 && (
+                  <Elem name="classifications-row">
+                    <Taxonomy
+                      selected={taxonomyPathsToSelectedItems(classifications?.default?.values)}
```

Comment:
This should be wrapped in a useMemo otherwise the underlying map will produce a new array each time the surrounding components would render, and force Taxonomy to also re-render.